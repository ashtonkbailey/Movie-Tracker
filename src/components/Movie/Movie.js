import React from 'react';
import '../../index.scss';


const Movie = ({ title, id, rating, text, release, poster }) => {
  return(
    <div className="movie">
      <img src={`https://image.tmdb.org/t/p/w500${poster}`} alt='' />
      <div className="movie-info">
        <div className="rating">
          <span>{rating.toFixed(1)}</span>
          <i className="fas fa-star" />
          </div>
        <h2>{title}</h2>
        <h5>{release}</h5>
        <p>{text}</p>
        <button className="add-to-favs">Add to Favorites</button>
      </div>
    </div>
  )
}

export default Movie;