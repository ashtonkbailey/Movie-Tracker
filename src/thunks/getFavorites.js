import { getFavorites } from '../actions/index'

export const getFavoritesThunk = (userId) => {
  return async (dispatch) => {
    try {
      const url = `http://localhost:3000/api/users/${userId}/favorites`
      const result = await fetch(url)
      if (!result.ok) {
        throw new Error('Unable to get your favorites')
      }
      const favorites = await result.json()
      const favoriteIds = favorites.data.map(favorite => favorite.movie_id)
      dispatch(getFavorites(favoriteIds))
    } catch (error) {
      console.log(error)
    }
  }
}
