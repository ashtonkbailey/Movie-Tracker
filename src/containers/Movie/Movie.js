import React, { Component } from 'react';
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import '../../index.scss';
import { addFavoriteThunk } from '../../thunks/addFavorite'
import { getFavoritesThunk } from '../../thunks/getFavorites'
import { removeFavoriteThunk } from '../../thunks/removeFavorite'

import PropTypes from 'prop-types';


export class Movie extends Component {
  
  handleAddFavorite = async () => {
    const { title, id, rating, text, release, poster, user, addFavoriteThunk, getFavoritesThunk, favorites } = this.props;

    let favoriteObj = {
      title: title,
      movie_id: id,
      vote_average: rating,
      overview: text,
      release_date: release,
      poster_path: poster,
      user_id: user.id
    }
    await addFavoriteThunk(favoriteObj, favorites)
    getFavoritesThunk(user.id)
  }

  handleRemoveFavorite = async () => {
    const { id, user, removeFavoriteThunk, getFavoritesThunk } = this.props;

    await removeFavoriteThunk(id, user.id)
    getFavoritesThunk(user.id)
  }

  render() {
    const { title, rating, text, release, poster, user, favorite } = this.props;

    let button
    if (!user.name) {
      button = (
        <Link to='/login'>
          <button className="add-to-favs">Login to Save</button>
        </Link>
      )
    } else if(favorite) {
      button = 
        <button
          onClick={()=>this.handleRemoveFavorite()}
          className="add-to-favs"
        >Remove Favorite</button>
    } else {
      button = 
        <button
          onClick={()=>this.handleAddFavorite()}
          className="add-to-favs"
        >Add to Favorites</button>
    }

    return(
      <div className={`movie ${favorite && 'favorite'}`}>
        <img src={`https://image.tmdb.org/t/p/w500${poster}`} alt='' />
        <div className="movie-info">
          <div className="rating">
            <span>{rating.toFixed(1)}</span>
            <i className="fas fa-star" />
            </div>
          <h2>{title}</h2>
          <h5>{release}</h5>
          <p>{text}</p>
          {button}
        </div>
      </div>
    )
  }
}

Movie.propTypes = {
  user: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  release: PropTypes.string.isRequired, 
  poster: PropTypes.string.isRequired,
  favorites: PropTypes.array.isRequired,
  addFavoriteThunk: PropTypes.func.isRequired,
  getFavoriteThunk: PropTypes.func.isRequired,
  removeFavoriteThunk: PropTypes.func.isRequired

}

export const mapStateToProps = (state) => ({
  user: state.user,
  favorites: state.favorites
})

export const mapDispatchToProps = (dispatch) => ({
  addFavoriteThunk: (favoriteObj, favorites) => dispatch(addFavoriteThunk(favoriteObj, favorites)),
  getFavoritesThunk: (userId) => dispatch(getFavoritesThunk(userId)),
  removeFavoriteThunk: (userId, movieId) => dispatch(removeFavoriteThunk(userId, movieId))
})

export default connect(mapStateToProps, mapDispatchToProps)(Movie);