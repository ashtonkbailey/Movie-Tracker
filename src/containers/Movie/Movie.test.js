import React from 'react';
import { mapStateToProps, mapDispatchToProps, Movie } from './Movie';
import { shallow } from 'enzyme'; 
import { addFavoriteThunk } from '../../thunks/addFavorite';
import { getFavoritesThunk } from '../../thunks/getFavorites';
import { removeFavoriteThunk } from '../../thunks/removeFavorite';


describe('Movie', () => {
  describe('Movie Component', () => {
    it('should match the snapshot when there is not a user logged in', () => {
      const mockUser = {}
      const mockFav = {
        title: 'MovieTitle',
        id: 123,
        rating: 9,
        text: 'MovieText',
        release: 'May4bewithyou'
      }
      const wrapper = shallow(<Movie user={mockUser} {...mockFav} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot when there is a user logged in and it is a favorite', () => {
      const mockUser = {name: ''}
      const mockFav = {
        title: 'MovieTitle',
        id: 123,
        rating: 9,
        text: 'MovieText',
        release: 'May4bewithyou',
        favorite: true
      }
      const wrapper = shallow(<Movie user={mockUser} {...mockFav} />)

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot when there is a user logged in and not a favorite', () => {
      const mockUser = {name: ''}
      const mockFav = {
        title: 'MovieTitle',
        id: 123,
        rating: 9,
        text: 'MovieText',
        release: 'May4bewithyou',
        favorite: false
      }
      const wrapper = shallow(<Movie user={mockUser} {...mockFav} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should parse the user and favorites from state', () => {
      let mockUser = {name: ''}
      let mockFav = {
        title: 'MovieTitle',
        id: 123,
        rating: 9,
        text: 'MovieText',
        release: 'May4bewithyou'
      }
      let mockFavoriteArray = [mockFav]
      let mockState = {
        favorites: mockFavoriteArray,
        user: mockUser,
        other: 'other thing in state'
      }
      const expectedFavorite = mockFavoriteArray
      const expectedUser = mockUser
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps.favorites).toEqual(expectedFavorite)
      expect(mappedProps.user).toEqual(expectedUser)
    })
  })

  describe('mapDispatchToProps', () => {

    it('should return a props object with the method addFavoriteThunk', () => {
      const favoriteObj = {
        title: 'MovieTitle',
        id: 123,
        rating: 9,
        text: 'MovieText',
        release: 'Maythe4thbewithyou'
      } 
      const favorites = []
      const mockDispatch = jest.fn()

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addFavoriteThunk(favoriteObj, favorites)
      mappedProps.getFavoritesThunk(2)
      mappedProps.removeFavoriteThunk(123, 2)

      expect(mockDispatch).toBeCalledTimes(3)
    })
  })

  describe('handleAddFavorite', () => {
    it('should addFavorite and fetch that update when handleAddFavorite is called', async () => {
      const mockProps = { title: 'A movie', id: 123, rating: 5, text: 'Some stuff', release: 'A Date', poster: 'pretty pictures', user: 1 }
      const mockAddThunk = jest.fn()
      const mockGetThunk = jest.fn()
      const mockUser = { name: 'Bob' }
      const wrapper = shallow(<Movie {...mockProps} addFavoriteThunk={mockAddThunk} getFavoritesThunk={mockGetThunk} user={mockUser} />)

      await wrapper.find('.add-to-favs').simulate('click')

      expect(mockAddThunk).toBeCalled()
      expect(mockGetThunk).toBeCalled()
    })
  })

  describe('handleRemoveFavorite', () => {
    it('should removeFavorite and fetch update when called', async () => {
      const mockProps = { title: 'A movie', id: 123, rating: 5, text: 'Some stuff', release: 'A Date', poster: 'pretty pictures', user: 1, favorite: true }
      const mockRemoveThunk = jest.fn()
      const mockGetThunk = jest.fn()
      const mockUser = { name: 'Bob' }
      const wrapper = shallow(<Movie {...mockProps} removeFavoriteThunk={mockRemoveThunk} getFavoritesThunk={mockGetThunk} user={mockUser} />)

      await wrapper.find('.add-to-favs').simulate('click')

      expect(mockRemoveThunk).toBeCalled()
      expect(mockGetThunk).toBeCalled()
    })
  })
})