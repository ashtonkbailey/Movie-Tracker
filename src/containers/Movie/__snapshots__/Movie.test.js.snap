// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Movie Component should match the snapshot when there is a user logged in and it is a favorite 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    favorite={true}
    id={123}
    rating={9}
    release="May4bewithyou"
    text="MovieText"
    title="MovieTitle"
    user={
      Object {
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="https://image.tmdb.org/t/p/w500undefined"
        />,
        <div
          className="movie-info"
        >
          <div
            className="rating"
          >
            <span>
              9.0
            </span>
            <i
              className="fas fa-star"
            />
          </div>
          <h2>
            MovieTitle
          </h2>
          <h5>
            May4bewithyou
          </h5>
          <p>
            MovieText
          </p>
          <Link
            replace={false}
            to="/login"
          >
            <button
              className="add-to-favs"
            >
              Login to Save
            </button>
          </Link>
        </div>,
      ],
      "className": "movie favorite",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "https://image.tmdb.org/t/p/w500undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="rating"
            >
              <span>
                9.0
              </span>
              <i
                className="fas fa-star"
              />
            </div>,
            <h2>
              MovieTitle
            </h2>,
            <h5>
              May4bewithyou
            </h5>,
            <p>
              MovieText
            </p>,
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="add-to-favs"
              >
                Login to Save
              </button>
            </Link>,
          ],
          "className": "movie-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  9.0
                </span>,
                <i
                  className="fas fa-star"
                />,
              ],
              "className": "rating",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "9.0",
                },
                "ref": null,
                "rendered": "9.0",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-star",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieTitle",
            },
            "ref": null,
            "rendered": "MovieTitle",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "May4bewithyou",
            },
            "ref": null,
            "rendered": "May4bewithyou",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieText",
            },
            "ref": null,
            "rendered": "MovieText",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-to-favs"
              >
                Login to Save
              </button>,
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login to Save",
                "className": "add-to-favs",
              },
              "ref": null,
              "rendered": "Login to Save",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="https://image.tmdb.org/t/p/w500undefined"
          />,
          <div
            className="movie-info"
          >
            <div
              className="rating"
            >
              <span>
                9.0
              </span>
              <i
                className="fas fa-star"
              />
            </div>
            <h2>
              MovieTitle
            </h2>
            <h5>
              May4bewithyou
            </h5>
            <p>
              MovieText
            </p>
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="add-to-favs"
              >
                Login to Save
              </button>
            </Link>
          </div>,
        ],
        "className": "movie favorite",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "https://image.tmdb.org/t/p/w500undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="rating"
              >
                <span>
                  9.0
                </span>
                <i
                  className="fas fa-star"
                />
              </div>,
              <h2>
                MovieTitle
              </h2>,
              <h5>
                May4bewithyou
              </h5>,
              <p>
                MovieText
              </p>,
              <Link
                replace={false}
                to="/login"
              >
                <button
                  className="add-to-favs"
                >
                  Login to Save
                </button>
              </Link>,
            ],
            "className": "movie-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    9.0
                  </span>,
                  <i
                    className="fas fa-star"
                  />,
                ],
                "className": "rating",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "9.0",
                  },
                  "ref": null,
                  "rendered": "9.0",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-star",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieTitle",
              },
              "ref": null,
              "rendered": "MovieTitle",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "May4bewithyou",
              },
              "ref": null,
              "rendered": "May4bewithyou",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieText",
              },
              "ref": null,
              "rendered": "MovieText",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-to-favs"
                >
                  Login to Save
                </button>,
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login to Save",
                  "className": "add-to-favs",
                },
                "ref": null,
                "rendered": "Login to Save",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Movie Movie Component should match the snapshot when there is a user logged in and not a favorite 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    favorite={false}
    id={123}
    rating={9}
    release="May4bewithyou"
    text="MovieText"
    title="MovieTitle"
    user={
      Object {
        "name": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="https://image.tmdb.org/t/p/w500undefined"
        />,
        <div
          className="movie-info"
        >
          <div
            className="rating"
          >
            <span>
              9.0
            </span>
            <i
              className="fas fa-star"
            />
          </div>
          <h2>
            MovieTitle
          </h2>
          <h5>
            May4bewithyou
          </h5>
          <p>
            MovieText
          </p>
          <Link
            replace={false}
            to="/login"
          >
            <button
              className="add-to-favs"
            >
              Login to Save
            </button>
          </Link>
        </div>,
      ],
      "className": "movie false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "https://image.tmdb.org/t/p/w500undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="rating"
            >
              <span>
                9.0
              </span>
              <i
                className="fas fa-star"
              />
            </div>,
            <h2>
              MovieTitle
            </h2>,
            <h5>
              May4bewithyou
            </h5>,
            <p>
              MovieText
            </p>,
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="add-to-favs"
              >
                Login to Save
              </button>
            </Link>,
          ],
          "className": "movie-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  9.0
                </span>,
                <i
                  className="fas fa-star"
                />,
              ],
              "className": "rating",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "9.0",
                },
                "ref": null,
                "rendered": "9.0",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-star",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieTitle",
            },
            "ref": null,
            "rendered": "MovieTitle",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "May4bewithyou",
            },
            "ref": null,
            "rendered": "May4bewithyou",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieText",
            },
            "ref": null,
            "rendered": "MovieText",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-to-favs"
              >
                Login to Save
              </button>,
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login to Save",
                "className": "add-to-favs",
              },
              "ref": null,
              "rendered": "Login to Save",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="https://image.tmdb.org/t/p/w500undefined"
          />,
          <div
            className="movie-info"
          >
            <div
              className="rating"
            >
              <span>
                9.0
              </span>
              <i
                className="fas fa-star"
              />
            </div>
            <h2>
              MovieTitle
            </h2>
            <h5>
              May4bewithyou
            </h5>
            <p>
              MovieText
            </p>
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="add-to-favs"
              >
                Login to Save
              </button>
            </Link>
          </div>,
        ],
        "className": "movie false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "https://image.tmdb.org/t/p/w500undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="rating"
              >
                <span>
                  9.0
                </span>
                <i
                  className="fas fa-star"
                />
              </div>,
              <h2>
                MovieTitle
              </h2>,
              <h5>
                May4bewithyou
              </h5>,
              <p>
                MovieText
              </p>,
              <Link
                replace={false}
                to="/login"
              >
                <button
                  className="add-to-favs"
                >
                  Login to Save
                </button>
              </Link>,
            ],
            "className": "movie-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    9.0
                  </span>,
                  <i
                    className="fas fa-star"
                  />,
                ],
                "className": "rating",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "9.0",
                  },
                  "ref": null,
                  "rendered": "9.0",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-star",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieTitle",
              },
              "ref": null,
              "rendered": "MovieTitle",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "May4bewithyou",
              },
              "ref": null,
              "rendered": "May4bewithyou",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieText",
              },
              "ref": null,
              "rendered": "MovieText",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-to-favs"
                >
                  Login to Save
                </button>,
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login to Save",
                  "className": "add-to-favs",
                },
                "ref": null,
                "rendered": "Login to Save",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Movie Movie Component should match the snapshot when there is not a user logged in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie
    id={123}
    rating={9}
    release="May4bewithyou"
    text="MovieText"
    title="MovieTitle"
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt=""
          src="https://image.tmdb.org/t/p/w500undefined"
        />,
        <div
          className="movie-info"
        >
          <div
            className="rating"
          >
            <span>
              9.0
            </span>
            <i
              className="fas fa-star"
            />
          </div>
          <h2>
            MovieTitle
          </h2>
          <h5>
            May4bewithyou
          </h5>
          <p>
            MovieText
          </p>
          <Link
            replace={false}
            to="/login"
          >
            <button
              className="add-to-favs"
            >
              Login to Save
            </button>
          </Link>
        </div>,
      ],
      "className": "movie undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "",
          "src": "https://image.tmdb.org/t/p/w500undefined",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="rating"
            >
              <span>
                9.0
              </span>
              <i
                className="fas fa-star"
              />
            </div>,
            <h2>
              MovieTitle
            </h2>,
            <h5>
              May4bewithyou
            </h5>,
            <p>
              MovieText
            </p>,
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="add-to-favs"
              >
                Login to Save
              </button>
            </Link>,
          ],
          "className": "movie-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  9.0
                </span>,
                <i
                  className="fas fa-star"
                />,
              ],
              "className": "rating",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "9.0",
                },
                "ref": null,
                "rendered": "9.0",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-star",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieTitle",
            },
            "ref": null,
            "rendered": "MovieTitle",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "May4bewithyou",
            },
            "ref": null,
            "rendered": "May4bewithyou",
            "type": "h5",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "MovieText",
            },
            "ref": null,
            "rendered": "MovieText",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="add-to-favs"
              >
                Login to Save
              </button>,
              "replace": false,
              "to": "/login",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login to Save",
                "className": "add-to-favs",
              },
              "ref": null,
              "rendered": "Login to Save",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt=""
            src="https://image.tmdb.org/t/p/w500undefined"
          />,
          <div
            className="movie-info"
          >
            <div
              className="rating"
            >
              <span>
                9.0
              </span>
              <i
                className="fas fa-star"
              />
            </div>
            <h2>
              MovieTitle
            </h2>
            <h5>
              May4bewithyou
            </h5>
            <p>
              MovieText
            </p>
            <Link
              replace={false}
              to="/login"
            >
              <button
                className="add-to-favs"
              >
                Login to Save
              </button>
            </Link>
          </div>,
        ],
        "className": "movie undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "",
            "src": "https://image.tmdb.org/t/p/w500undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="rating"
              >
                <span>
                  9.0
                </span>
                <i
                  className="fas fa-star"
                />
              </div>,
              <h2>
                MovieTitle
              </h2>,
              <h5>
                May4bewithyou
              </h5>,
              <p>
                MovieText
              </p>,
              <Link
                replace={false}
                to="/login"
              >
                <button
                  className="add-to-favs"
                >
                  Login to Save
                </button>
              </Link>,
            ],
            "className": "movie-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    9.0
                  </span>,
                  <i
                    className="fas fa-star"
                  />,
                ],
                "className": "rating",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "9.0",
                  },
                  "ref": null,
                  "rendered": "9.0",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "fas fa-star",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieTitle",
              },
              "ref": null,
              "rendered": "MovieTitle",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "May4bewithyou",
              },
              "ref": null,
              "rendered": "May4bewithyou",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "MovieText",
              },
              "ref": null,
              "rendered": "MovieText",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="add-to-favs"
                >
                  Login to Save
                </button>,
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login to Save",
                  "className": "add-to-favs",
                },
                "ref": null,
                "rendered": "Login to Save",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
